### PR TITLE
Random Num Gen Weighted Int and distributed Float 

### PR DESCRIPTION
**Pull request by vicdoval**
## Addressed problem description
"Random Num Gen" offers uniform distributions, but it could be expanded (using numpy) it to create Weighted distribution of Integers and non-uniform distribution of floats

## Solution description

I added most of the methods described in de numpy.random reference https://docs.scipy.org/doc/numpy-1.14.0/reference/routines.random.html

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

**Comment by vicdoval** [comment]
Images for docs
screenrandom
screenrandom1
screenrandom2
screenrandom3

**Comment by zeffii** [comment]
nice work!  but i'm a little uncomfortable with the retention of of the classname. an MK2 seems more appropriate.

another small issue i have is with adding/remove sockets. please look at examples of using `.hide_safe`

**Comment by vicdoval** [comment]
Thanks! 

I will change the `.hide_safe`,  it looks great
About the MK2 thing:  with the updates it  does not break old setups as the default is the same as how it use to be (Uniform distribution)
¿Any way would you be fine with changing the class but leaving the name un-touched?
(I don't like very much the MK2 on the menu)

**Comment by vicdoval** [comment]
Also not changing the class is more like a automatic update 😃

**Comment by Durman** [comment]
Great.)
test1
test2

**Comment by vicdoval** [comment]
long life to .hide_self 😃

**Comment by zeffii** [comment]
i'm more confident about this Pull Request now :)

**Comment by zeffii** [comment]
code looks pretty nice btw.

File: nodes/number/random_num_gen.py [hunk 1]
```diff
@@ -27,7 +28,11 @@
 
 
 class SvRndNumGen(bpy.types.Node, SverchCustomTreeNode):
-    ''' Generate a random number (int of float) thru a given range (inclusive) '''
+    """
+    Triggers: Random thru a range
+    Tooltip: Generate a random number (int of float) thru a given range (inclusive) .
+    """
+    ''' '''
```
Comment (zeffii) [inline nodes/number/random_num_gen.py]: empty string?

**Comment by vicdoval** [comment]
> code looks pretty nice btw.

I've been learning from the best (the sverchok developers) 😃

**Comment by zeffii** [comment]
good man!

**Pull request merged**